File: HPWorkpath_v1.6.3/Samples/ExampleAPIServices/source/MassStorageSample/src/com/hp/workpath/sample/massstoragesample/task/DeleteFileTask.java
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.massstoragesample.task;

import android.os.Handler;
import android.os.Looper;
import android.view.View;

import com.hp.workpath.api.massstorage.CustomerDataFile;
import com.hp.workpath.sample.massstoragesample.Logger;
import com.hp.workpath.sample.massstoragesample.MainActivity;
import com.hp.workpath.sample.massstoragesample.R;

import java.lang.ref.WeakReference;
import java.util.List;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

public class DeleteFileTask {

    private final WeakReference<MainActivity> mContextRef;
    private final List<CustomerDataFile> mFilesToBeDeleted;
    private final CustomerDataFile mBaseCustomerDataFile;
    private int failCount = 0;
    private int successCount = 0;

    private ExecutorService executor = Executors.newSingleThreadExecutor();
    private Handler handler = new Handler(Looper.getMainLooper());

    public DeleteFileTask(MainActivity context, List<CustomerDataFile> fileList, CustomerDataFile baseCustomerDataFile) {
        this.mContextRef = new WeakReference<>(context);
        this.mFilesToBeDeleted = fileList;
        this.mBaseCustomerDataFile = baseCustomerDataFile;
    }

    public void taskExecute() {
        try {
            executor.execute(new Runnable() {
                @Override
                public void run() {
                    try {
                        for (CustomerDataFile customerDataFile : mFilesToBeDeleted) {
                            boolean isDeleted = customerDataFile.delete();
                            if (isDeleted) {
                                ++successCount;
                            } else {
                                ++failCount;
                            }
                        }
                        onPostExecute(true);
                    } catch (Throwable t) {
                        Logger.showResult(null, "CustomerDataFile.delete " + t.getMessage());
                        executor.shutdown();
                        onPostExecute(false);
                    }
                }
            });
        } catch (Exception e) {
            Logger.showResult(null, "CustomerDataFile.delete " + e.getMessage());
            onPostExecute(false);
            executor.shutdown();
        }
    }

    private void onPostExecute(Boolean result) {
        handler.post(new Runnable() {
            @Override
            public void run() {
                mContextRef.get().enableButton(true);
                mContextRef.get().showProgress(View.GONE);
                Logger.showResult(mContextRef.get(), mContextRef.get().getString(R.string.sucess_and_fail, successCount, failCount));
                if (result) {
                    mContextRef.get().displayFileList(mBaseCustomerDataFile);
                }
            }
        });
    }
}

File: HPWorkpath_v1.6.3/Samples_Kotlin/ExampleAPIServices/source/MassStorageSample/src/com/hp/workpath/sample/massstoragesample/task/DeleteFileTask.kt
// Copyright 2018 HP Development Company, L.P.
// SPDX-License-Identifier: MIT
package com.hp.workpath.sample.massstoragesample.task

import android.view.View
import com.hp.workpath.api.massstorage.CustomerDataFile
import com.hp.workpath.sample.massstoragesample.Logger
import com.hp.workpath.sample.massstoragesample.MainActivity
import com.hp.workpath.sample.massstoragesample.R
import java.lang.ref.WeakReference

class DeleteFileTask(context: MainActivity, fileList: List<CustomerDataFile>, baseCustomerDataFile: CustomerDataFile){

    private val mContextRef: WeakReference<MainActivity> = WeakReference(context)
    private val mFilesToBeDeleted: List<CustomerDataFile> = fileList
    private val mBaseCustomerDataFile: CustomerDataFile = baseCustomerDataFile
    private var mThrowable: Throwable? = null
    private var failCount = 0
    private var successCount = 0

    suspend fun execute() {
        var result = false
        try {
            for (customerDataFile in mFilesToBeDeleted) {
                val isDeleted = customerDataFile.delete()
                if (isDeleted) {
                    ++successCount
                } else {
                    ++failCount
                }
            }
            result = true
        } catch (t: Throwable) {
            mThrowable = t
        }
        onPostExecute(result)
    }

    private suspend fun onPostExecute(result: Boolean) {
        mContextRef.get()?.run {
            runOnUiThread {
                enableButton(true)
                showProgress(View.GONE)
                Logger.showResult(this, getString(R.string.sucess_and_fail, successCount, failCount))
                if (result) {
                    displayFileList(mBaseCustomerDataFile)
                } else {
                    if (mThrowable != null) {
                        Logger.showResult(mContextRef.get(), "CustomerDataFile.delete ${mThrowable?.message}")
                    }
                }
            }
        }
    }
}